Fix bug in emitUse to drop all values that are marked as drop

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56079 Enable forward/backward compatibility in TS mobile
* **#56652 Fix bug in emitUse to drop all values that are marked as drop**

Summary: Previous code doesn't drop prim::Constant values even when they are marked as drop.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27927413](https://our.internmc.facebook.com/intern/diff/D27927413)

tugsbayasgalan committed:
Fix bug in emitUse to drop all values that are marked as drop

Summary: Previous code doesn't drop prim::Constant values even when they are marked as drop.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix bug in emitUse to drop all values that are marked as drop"


Summary: Previous code doesn't drop prim::Constant values even when they are marked as drop.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27927413](https://our.internmc.facebook.com/intern/diff/D27927413)

[ghstack-poisoned]

iseeyuan commented on torch/csrc/jit/runtime/interpreter.cpp:
```diff
@@ -617,13 +617,15 @@ struct CodeImpl {
       }
+
+      if (drop) {
+        op = DROPR;
```
It's DROPR, which clears the corresponding register. It does not pop stack. In such case, would the stack still holds the default arg?

tugsbayasgalan committed:
Update on "Fix bug in emitUse to drop all values that are marked as drop"


Summary: Previous code doesn't drop prim::Constant values even when they are marked as drop.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27927413](https://our.internmc.facebook.com/intern/diff/D27927413)

[ghstack-poisoned]
